plugin.py

```diff
@@ -155,7 +152,9 @@ def m_dart_textDocument_publishClosingLabels(self, params: Any) -> None:
             except AttributeError:
                 phantom_set = sublime.PhantomSet(sv.view, self.phantom_key)
                 setattr(sv, "_lsp_dart_labels", phantom_set)
-            phantom_set.update(self.closing_labels(sv.view, reversed(params["labels"])))
+            closing_labels = self.closing_labels(sv.view, reversed(params["labels"]))
+            if closing_labels:
```

Comment:
Is this if necessary?

Lets say that the server published one `publishClosingLabels` notification
and that the server returned one label `[{'range': {},'label': 'AppBar' }]` 

than we selected the whole view(ctrl+a) and deleted it, 
the server would probably return an empty array `[]`.

if we have this if check (an empty `[]` is falsy, and we never updated the phantoms), 
than we would still end up seeing the previous phantoms,
but we would expect to see none,  

(I haven't setup flutter, I don't think that I will have the time to set it up today).

Reply:
The point being: maybe we should always update the phantoms to never end up in a case where we show stale phantoms from before.

```python
closing_labels = self.closing_labels(sv.view, reversed(params["labels"]))
phantom_set.update(closing_labels)          # or 
# phantom_set.update(closing_labels or [])  # in case closing_labels could be None
```

Reply:
just tested your scenario, the phantoms are deleted after I `ctrl+a del`, but yea I agree that if the server sends for whatever reason `[]`, phantoms wont be deleted, I'll make the changes